lib/flash.toit

```diff
@@ -0,0 +1,110 @@
+// Copyright (C) 2020 Toitware ApS. All rights reserved.
+// Use of this source code is governed by an MIT-style license that can be
+// found in the lib/LICENSE file.
+
+import spi
+import reader show Reader
+import writer show Writer
+import gpio
+
+
+FLASH_5MHZ    ::= 0 ///< The flash runs under 5MHz
+FLASH_10MHZ   ::= 1 ///< The flash runs under 10MHz
+FLASH_20MHZ   ::= 2 ///< The flash runs under 20MHz
+FLASH_26MHZ   ::= 3 ///< The flash runs under 26MHz
+FLASH_40MHZ   ::= 4 ///< The flash runs under 40MHz
+FLASH_80MHZ   ::= 5 ///< The flash runs under 80MHz
+
+class Flash:
+  flash_ := null
+  mount_point/string
+
+  /**
+  Mounts an sdcard as a FAT file system under $mount_point on the $spi_bus.
+
+  The $cs is the chip select pin for the sdcard holder and $frequency is the spi frequency.
+
+  If $format is true, then format the sdcard with $max_files and $allocation_unit_size if it is not formatted.
+  */
+  constructor.sdcard
+      --.mount_point/string
+      --spi_bus/spi.Bus
+      --cs/gpio.Pin
+      --format/bool=false
+      --max_files/int=5
+      --allocation_unit_size/int=16384:
+    if not mount_point.starts_with "/": throw "INVALID_ARGUMENT"
+    flash_ = init_sdcard_ mount_point spi_bus.spi_ cs.num (format?1:0) max_files allocation_unit_size
```

Comment:
I think it's simpler to just pass the boolean to the primitive.

Would it make sense to have a separate constructor for the format case? The `max_files` and `allocation_unit_size` only makes sense when `format` is true, which is an argument for a separate constructor, no? You can still use the same primitive.